action/protocol/account/protocol.go

```diff
@@ -80,6 +82,24 @@ func (p *Protocol) Validate(ctx context.Context, act action.Action, sr protocol.
 			return errors.Wrap(err, "error when validating transfer action")
 		}
 	}
+	// check balance of sender
+	if !protocol.MustGetFeatureCtx(ctx).SufficentBalanceGuarantee {
```

Comment:
i think this is needed by all protocols, like put somewhere in `workingset`?
transfer protocol already has the check inside `Handle()`

Reply:
moved to genericValidator, which will be used at ValidateBlock